src/intersections/boxes.jl

```diff
@@ -34,3 +34,51 @@ function intersecttype(f::Function, b1::Box{Dim,T}, b2::Box{Dim,T}) where {Dim,T
     return FaceTouchingBoxes(Box(u, v)) |> f
   end
 end
+
+"""
+    intersecttype(function, ray, box)
+
+Calculate the intersection type of a `ray`` and a `box`` and apply function `f` to it.
+
+## References
+
+* Williams A, Barrus S, Morley R K, et al., 2005. [An efficient and robust ray-box
+  intersection algorithm]
+  (https://dl.acm.org/doi/abs/10.1145/1198555.1198748)
+"""
+function intersecttype(f::Function, r::Ray{Dim,T}, b::Box{Dim,T}) where {Dim,T}
+  invdir = one(T) ./ direction(r)
+  lo, up = coordinates.(extrema(b))
+  orig = coordinates(origin(r))
+
+  tmin = zero(T)
+  tmax = typemax(T)
```

Comment:
I find this asymmetry mathematically unpleasant. Are you sure that the minimum should be initialized as zero and the maximum as typemax? Shouldn't the minimum be initialized as typemin? Can you please justify this discrepancy?

Reply:
@juliohm Because a ray is asymmetry. A ray can not extend on one direction (follows it's direction vector). Here t describes the distance between the intersection and the starting point. The minimum distance is zero instead of -Inf. If I use typemin, it means allowing the ray to extend in the reverse direction.